### PR TITLE
[CUTEst.jl] Support Windows

### PR DESCRIPTION
**Pull request by amontoison**
:crossed_fingers: 
Should be merged after #344 and #349 if it works one day...

### CODE REVIEW

**Comment by amontoison** [comment]
Wow, it works :flushed: :astonished: :scream: 
cc @dpo

File: src/CUTEst.jl [hunk 1]
```diff
@@ -28,12 +28,14 @@ const cutest_problems_path = joinpath(dirname(@__FILE__), "..", "deps", "files")
 isdir(cutest_problems_path) || mkpath(cutest_problems_path)
 
 function __init__()
```
Comment (tmigot) [inline src/CUTEst.jl:30]:
```suggestion
function __init__()
```
Could you just add a comment around to say why we don't do anything for Windows here?

**Review by tmigot — CHANGES_REQUESTED** [review]
That's really cool @amontoison, good job!
I am assuming this wouldn't work for anyone just testing directly on windows though, so we need a bit of documentation at least to warn them.

File: src/CUTEst.jl (hunk 1 above)
Comment (amontoison) [reply src/CUTEst.jl:30]: It's because we don't have `bash` on Windows. I will add a comment in a different PR.

**Comment by amontoison** [comment]
> That's really cool @amontoison, good job! I am assuming this wouldn't work for anyone just testing directly on windows though, so we need a bit of documentation at least to warn them.

Yes, we require the user to have a Fortran compiler `gfortran` for now but I know how to install it as an artifact for users so it will be very user-friendly in the future. I will reuse the artifacts of `Artifacts.toml` from PackageCompiler.jl.

**Pull request merged**